pkg/api/v1alpha1/cluster.go

```diff
@@ -849,8 +850,8 @@ func validateMDUpgradeRolloutStrategy(w *WorkerNodeGroupConfiguration) error {
 		return nil
 	}
 
-	if w.UpgradeRolloutStrategy.Type != "RollingUpdate" {
-		return fmt.Errorf("WorkerNodeGroupConfiguration: only 'RollingUpdate' supported for upgrade rollout strategy type")
+	if w.UpgradeRolloutStrategy.Type != "RollingUpdate" && w.UpgradeRolloutStrategy.Type != "InPlace" {
+		return fmt.Errorf("WorkerNodeGroupConfiguration: only 'RollingUpdate' and 'InPlace' are supported for upgrade rollout strategy type")
```

Comment:
This looks like a provider agnostic check. So looks like we might need to restrict "InPlace"  at other providers once the rolloutStrategy support PR for other providers get merged in.? Another option is to check if the type is inplace and `if cpUpgradeRolloutStrategy.Type == "InPlace" && cluster.Spec.DatacenterRef.Kind != TinkerbellDatacenterKind` and error out for this case here.

Reply:
Yes, I'll include the logic to restrict this feature for the other providers in the validations PR.